tests/test_gen_artifacts.py

```diff
@@ -0,0 +1,83 @@
+import stat
+from pathlib import Path
+
+import click
+import click.testing
+
+
+from exhibitor_tls_artifacts.gen_artifacts import app
+
+
+class TestCLI:
+    @staticmethod
+    def _validate_files(directory):
+        """ Ensures that artifacts are created and that they have the
+        appropriate permission """
+        files = [
+            ('client-cert.pem', 0o100644),
+            ('client-key.pem', 0o100600),
+            ('clientstore.jks', 0o100600),
+            ('root-cert.pem', 0o100644),
+            ('serverstore.jks', 0o100600),
+            ('truststore.jks', 0o100600)
+        ]
+        for f, mode in files:
+            full_path = directory / f
+            assert full_path.exists()
+            assert full_path.stat()[stat.ST_MODE] == mode
+
+    def test_default(self):
+        """ Tests the most basic operation """
```

Comment:
No need to change but just FYI https://jml.io/pages/test-docstrings.html - that's a great (short!) guide for test docstrings that I prefer to follow.